[Bug]: failed to compile milvus due to in-compatible zstd-sys

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

[2024-03-28T06:16:28.059Z] error[E0432]: unresolved import `zstd_sys::ZSTD_cParameter::ZSTD_c_experimentalParam6`

[2024-03-28T06:16:28.059Z]    --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/zstd-safe-6.0.6/src/lib.rs:609:13

[2024-03-28T06:16:28.059Z]     |

[2024-03-28T06:16:28.059Z] 609 |             ZSTD_c_experimentalParam6 as ZSTD_c_targetCBlockSize,

[2024-03-28T06:16:28.059Z]     |             -------------------------^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2024-03-28T06:16:28.059Z]     |             |

[2024-03-28T06:16:28.059Z]     |             no `ZSTD_c_experimentalParam6` in `ZSTD_cParameter`

[2024-03-28T06:16:28.059Z]     |             help: a similar name exists in the module: `ZSTD_c_experimentalParam1`

[2024-03-28T06:16:28.059Z] 

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

--- a/internal/core/thirdparty/tantivy/tantivy-binding/Cargo.toml
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/Cargo.toml
@@ -6,10 +6,11 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-tantivy = "0.21.0"
+tantivy = "=0.21.1"
 futures = "0.3.21"
 libc = "0.2"
 scopeguard = "1.2"
+zstd-sys = "=2.0.9"
 
 [build-dependencies]
 cbindgen = "0.26.0"

